actions/git-hooks/pre-push

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+#
+
+# When run as a hook by git, DIR be the repository's root folder.
+DIR=$(pwd)
+
+PREFIX="[pre-commit]"
+
+echo "$PREFIX Running hook..."
+
+#
+# CHECK FOR RUNNING CONTAINERS
+#
+echo "$PREFIX Checking docker-compose..."
+cd artifacts || exit
+if [ -w "$(which docker-compose)" ]; then
+	DC="docker-compose ps"
+else
+	DC="sudo docker-compose ps"
+fi
+CONTAINERS=('hydra_1' 'hydra_db_1' 'hydra_migrate' 'mongo-express' 'mongo_1' 'redis')
+for c in "${CONTAINERS[@]}"; do
+	if  ! (eval "$DC" | grep -q "$c" &> /dev/null) ; then
+		echo "$PREFIX Please ensure docker-compose is up and running."
+		exit 1
+fi
+done
+echo "$PREFIX All containers are up and running!"
+
+#
+# RUN INTEGRATION TESTS
+#
+echo "$PREFIX Running IAM integration tests..."
+
+cd "$DIR/api/pkg/apps/iam" || exit
+export GO111MODULE=on
+if ! go test; then
+	>&2 echo "[pre-push] Integration tests failed, see output above."
+	exit 1
+fi
+
+echo "$PREFIX Running CMS integration tests..."
+
+cd "$DIR/api/pkg/apps/cms" || exit
+if ! go test; then
+	>&2 echo "[pre-push] Integration tests failed, see output above."
+	exit 1
+fi
+
+#
+# RUN E2E TESTS
+#
+
+# check for localhost:9000 listening
+#
+echo "$PREFIX Checking webapp"
+serverIsRunning() {
+	netstat -tln | grep -q ":9000" &> /dev/null
+	return $?
+}
+if ! serverIsRunning; then
+	echo "$PREFIX webapp is not running, attempting to start it now."
+	cd "$DIR/api" || exit
+	make serve &> /dev/null &
+
+	n=0
+  until [ "$n" -ge 5 ]
+  do
+     serverIsRunning && break
+     n=$((n+1))
+     sleep 10
+  done
+
+	if ! serverIsRunning; then
+		>&2 echo "$PREFIX Unable to start webapp."
+		exit 1
+	fi
+fi
+echo "$PREFIX webapp is up and running!"
+
+cd "$DIR/e2e" || exit
+
+# linting
+#
+echo "$PREFIX Running eslint on .js files..."
+if ! npm run lint:fix; then
+	>&2 echo "[pre-push] Linting failed, see output above."
+	exit 1
+fi
+
+echo "$PREFIX Running E2E tests..."
```

Comment:
maybe we can skip the e2e until the new design comes in and we fix the broken tests?

Reply:
I opted to skip the invalid tests themselves, that way we at least have coverage for some of the pages